### PR TITLE
Override equals method for CommandDataImpl and related types

### PR DESCRIPTION
**Pull request by Kaktushose**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [x] Internal code
- [ ] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: 

Closes Issue: NaN

## Description

Overrides the equals methods of `CommandDataImpl`, `SubcommandData`, `OptionData` and `SubcommandGroupData`.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/api/interactions/commands/build/SubcommandData.java [hunk 1]
```diff
@@ -362,4 +362,19 @@ public static SubcommandData fromSubcommand(@Nonnull Command.Subcommand subcomma
...
+    public boolean equals(Object o)
+    {
+        if (this == o)
+            return true;
+        if (!(o instanceof SubcommandData))
+            return false;
+        SubcommandData that = (SubcommandData) o;
+        return allowRequired == that.allowRequired &&
+                options.stream(DataArray::getObject).map(OptionData::fromData).collect(Collectors.toList())
+                        .equals(options.stream(DataArray::getObject).map(OptionData::fromData).collect(Collectors.toList())) &&
```
Comment (freya022) [inline src/main/java/net/dv8tion/jda/api/interactions/commands/build/SubcommandData.java:376]: Those are the same lines

File: src/main/java/net/dv8tion/jda/api/interactions/commands/build/SubcommandGroupData.java [hunk 2]
```diff
@@ -249,4 +249,18 @@ public static SubcommandGroupData fromGroup(@Nonnull Command.SubcommandGroup gro
...
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) return true;
+
+        if (!(o instanceof SubcommandGroupData))
+            return false;
+        SubcommandGroupData that = (SubcommandGroupData) o;
+        return options.stream(DataArray::getObject).map(OptionData::fromData).collect(Collectors.toList())
+                .equals(options.stream(DataArray::getObject).map(OptionData::fromData).collect(Collectors.toList())) &&
```
Comment (freya022) [inline src/main/java/net/dv8tion/jda/api/interactions/commands/build/SubcommandGroupData.java:262]: Same lines too

File: src/main/java/net/dv8tion/jda/internal/interactions/CommandDataImpl.java [hunk 3]
```diff
@@ -255,4 +255,24 @@ public List<OptionData> getOptions()
...
+        if (!(o instanceof CommandDataImpl))
+            return false;
+        CommandDataImpl that = (CommandDataImpl) o;
+        return allowSubcommands == that.allowSubcommands &&
+                allowGroups == that.allowGroups &&
+                allowOption == that.allowOption &&
+                defaultPermissions == that.defaultPermissions &&
+                allowRequired == that.allowRequired &&
+                options.stream(DataArray::getObject).map(OptionData::fromData).collect(Collectors.toList())
+                        .equals(options.stream(DataArray::getObject).map(OptionData::fromData).collect(Collectors.toList())) &&
```
Comment (freya022) [inline src/main/java/net/dv8tion/jda/internal/interactions/CommandDataImpl.java:273]: Again

**Comment by Kaktushose** [comment]
So the problem is that the order of options you get from `retrieveCommands` can differ from the inital order when the `SlashCommandData` was created. 

That's why I changed all `a.equals(b)` to `a.containsAll(b) && a.size() == b.size()`. Not sure if this is what we want though

**Pull request closed without merging**